assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -351,7 +354,7 @@ angular.module('openshiftConsole')
           $scope.imageSourceBuildFrom.tags = {};
           var projectImageStreams = imageStreams.by("metadata.name");
           if (!_.isEmpty(projectImageStreams)) {
-            if (!Object.keys(projectImageStreams).contains($scope.imageSourceBuildFrom.imageStream) && projectName === $scope.imageSourceBuildFrom.namespace) {
+            if (!Object.keys(projectImageStreams).contains($scope.imageSourceBuildFrom.imageStream) && projectName === $scope.imageSourceBuildFrom.namespace && !selectFirstOption) {
```

Comment:
These should really use `Object.hasOwnProperty` or `_.has`.

```
!_.has(projectImageStreams, $scope.imageSourceBuildFrom.imageStream)
```

Not new in this PR, but worth fixing since we're changing the code.